tensorpack/tfutils/symbolic_functions.py

```diff
@@ -212,3 +212,142 @@ def saliency_map(output, input, name="saliency_map"):
     saliency_op = tf.gradients(max_outp, input)[:][0]
     saliency_op = tf.identity(saliency_op, name=name)
     return saliency_op
+
+
+def l2_norm(t, eps=1e-12):
+    """return L2 norm for input x
```

Comment:
I don't think reducing on axis=1 (not axis=None, not axis=[1,2,...end], but axis=1) is what people would expect on a function named `l2_norm`. And it would also take more documentation to explain that it reduces on axis=1, than writing the code itself.
Could you just put this function where it is used?